src/hugr/validate.rs

```diff
@@ -158,7 +159,16 @@ impl<'a, 'b> ValidationContext<'a, 'b> {
             }
         }
 
-        // Check operation-specific constraints
+        // Check operation-specific constraints. Firstly that type args are correct
+        // (note that if the Hugr were mutable here, this could be a good time to do
+        // resolve_extension_ops)
```

Comment:
This refers to #508 so maybe isn't a good comment to leave in the code...thoughts welcome here and/or on that issue as appropriate

Reply:
I'd say it's useful to keep the read-only validation from any inference/resolution/etc mutation.

In any case, I'd either remove this comment since we have the open Issue, or at least link to the issue from here.